tests/staggered_gsmear_test_utils.h

```diff
@@ -0,0 +1,405 @@
+      //
+      gauge_param.type     = QUDA_ASQTAD_LONG_LINKS;
+      gauge_param.location = QUDA_CPU_FIELD_LOCATION;
+
+      GaugeFieldParam cpuTwoLinkParam(gauge_param, milc_twolnk);
+      cpuTwoLinkParam.ghostExchange = QUDA_GHOST_EXCHANGE_PAD;
+      cpuTwoLink                    = GaugeField::Create(cpuTwoLinkParam);    
+    
+      ColorSpinorParam cs_param;
+
+      constructStaggeredTestSpinorParam(&cs_param, &inv_param, &gauge_param);
+
+      spinor    = std::make_unique<ColorSpinorField>(cs_param);
+      spinorRef = std::make_unique<ColorSpinorField>(cs_param);
+      tmp       = std::make_unique<ColorSpinorField>(cs_param);
+      tmp2      = std::make_unique<ColorSpinorField>(cs_param);    
+
+      spinor->Source(QUDA_RANDOM_SOURCE);
+    
+      *tmp = *spinor;
+    }
+  }
+
+  void end()
+  {
+    // Clean up gauge fields
+    for (int dir = 0; dir < 4; dir++) {
+      host_free(qdp_inlink[dir]);
+      host_free(qdp_inlink_ex[dir]);
+      host_free(qdp_twolnk[dir]);
+      host_free(qdp_ref_twolnk[dir]);
+    }
+  
+    host_free(milc_inlink);
+    host_free(milc_twolnk);
+
+    if (cpuTwoLink != nullptr) { delete cpuTwoLink; cpuTwoLink = nullptr; }
+
+    if(gtest_type == gsmear_test_type::GaussianSmear) {
+      tmp2.reset();
+      tmp.reset();
+      spinor.reset();
+      spinorRef.reset();
+    }
+
+    freeGaugeQuda();
+
+    commDimPartitionedReset();
+  }
+
+  GSmearTime gsmearCUDA(int niter) //niter
+  {
+    GSmearTime gsmear_time;
+
+    host_timer_t host_timer;
+    device_timer_t device_timer;
+
+    comm_barrier();
+    device_timer.start();
+
+    printfQuda("running test in %d iters.", niter);
+
+    // smearing parameters
+    switch (gtest_type) {
+      case gsmear_test_type::TwoLink:
+      {
+        printfQuda("Doing two link in QUDA\n");
+        computeTwoLinkQuda((void*) milc_twolnk, nullptr, &gauge_param);
+	break;
+      }
+      case gsmear_test_type::GaussianSmear:
+      {
+        double omega       = 2.0;
+        int n_steps        = 50;
+        double smear_coeff = -1.0 * omega * omega / ( 4*n_steps );
+
+        const int compute_2link = 1;
+        const int delete_2link  = 0;
+        const int t0            = -1;
+  
+        performTwoLinkGaussianSmearNStep(spinor->V(), &inv_param, n_steps, smear_coeff, compute_2link, delete_2link, t0);
+
+        break;
+      }
+      default: errorQuda("Test type not defined");
+    }
+
+    device_timer.stop();
+    gsmear_time.event_time = device_timer.last();
+
+    return gsmear_time;
+  }
+
+  void run_test(int niter, bool print_metrics = false)
+  {
+    printfQuda("Tuning...\n");
+    gsmearCUDA(1);
+
+
+    GSmearTime gsmear_time = gsmearCUDA(niter);
+    if(gtest_type == gsmear_test_type::GaussianSmear) *spinorRef = *spinor;
+
+    if (print_metrics) {
+      printfQuda("%fus per kernel call\n", 1e6 * gsmear_time.event_time / niter);
+#if 0
```

Comment:
Is the code below going to enabled at some point?

Reply:
done